File: src/#894/solution.go
package solution

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func allPossibleFBT(N int) []*TreeNode {
	if N%2 == 0 {
		return []*TreeNode{}
	}
	if N == 1 {
		return []*TreeNode{&TreeNode{0, nil, nil}}
	}
	res := []*TreeNode{}
	for i := 1; i < N; i += 2 {
		L := allPossibleFBT(i)
		R := allPossibleFBT(N - i - 1)
		for _, l := range L {
			for _, r := range R {
				res = append(res, &TreeNode{0, l, r})
			}
		}
	}
	return res
}

File: src/#894/solution.js
function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val
  this.left = left === undefined ? null : left
  this.right = right === undefined ? null : right
}

var allPossibleFBT = function (N) {
  if (N % 2 === 0) return []
  if (N === 1) return [new TreeNode(0, null, null)]

  let res = []
  for (let i = 1; i < N; i++) {
    let L = allPossibleFBT(i)
    let R = allPossibleFBT(N - i - 1)
    L.forEach(l => {
      R.forEach(r => {
        let root = new TreeNode(0, l, r)
        res.push(root)
      })
    })
  }
  
  return res
}
